src/contraction_hierarchies/src/libch.cpp

```diff
@@ -304,6 +304,12 @@ inline ostream& operator<< (ostream& os, const Edge& e) {
             poiIndexArray.push_back(CHPOIIndex(this->staticGraph, maxDistanceToConsider, maxNumberOfPOIsInBucket, numberOfThreads));
         }
     }
+
+    void ContractionHierarchies::createPOIIndex(unsigned categoryNum, unsigned maxDistanceToConsider, unsigned maxNumberOfPOIsInBucket){
+        CHASSERT(this->staticGraph != NULL, "Preprocessing not finished");
+        // reinitialize this bucket
+        poiIndexArray[categoryNum] = CHPOIIndex(this->staticGraph, maxDistanceToConsider, maxNumberOfPOIsInBucket, numberOfThreads);
```

Comment:
I think it's a good idea to check if the categoryNum is bigger than poiIndexArray size (given that's a std::vector), just in case.